packages/frontend/src/pages/Stake/Stake.tsx

```diff
@@ -1,115 +1,148 @@
-import React, { FC, useMemo } from 'react'
-import { makeStyles } from '@material-ui/core/styles'
+import React, { FC, useEffect, useMemo, useState } from 'react'
 import Box from '@material-ui/core/Box'
 import Typography from '@material-ui/core/Typography'
-import { Token, CanonicalToken, WrappedToken } from '@hop-protocol/sdk'
+import { Token, CanonicalToken, WrappedToken, ChainId, ChainSlug } from '@hop-protocol/sdk'
 import { useApp } from 'src/contexts/AppContext'
 import { useWeb3Context } from 'src/contexts/Web3Context'
 import StakeWidget from 'src/pages/Stake/StakeWidget'
 import useAsyncMemo from 'src/hooks/useAsyncMemo'
+import { useSelectedNetwork } from 'src/hooks'
 import { isMainnet } from 'src/config'
-import { Flex } from 'src/components/ui'
-import { findMatchingBridge } from 'src/utils'
+import { Div, Flex } from 'src/components/ui'
+import { findMatchingBridge, findNetworkBySlug, getNetworkProviderOrDefault } from 'src/utils'
 import { StakingRewards__factory } from '@hop-protocol/core/contracts'
+import { providers, Signer } from 'ethers'
+import Network from 'src/models/Network'
+import { RaisedNetworkSelector } from 'src/components/NetworkSelector/RaisedNetworkSelector'
 
-const useStyles = makeStyles(theme => ({
-  container: {
-    padding: `${theme.padding.default} 0`,
+const stakingRewardsContracts = {
+  polygon: {
+    ETH: '0x7bCeDA1Db99D64F25eFA279BB11CE48E15Fda427',
+    MATIC: '0x7dEEbCaD1416110022F444B03aEb1D20eB4Ea53f',
+    USDC: '0x2C2Ab81Cf235e86374468b387e241DF22459A265',
+    USDT: '0x07932e9A5AB8800922B2688FB1FA0DAAd8341772',
+  },
+  gnosis: {
+    DAI: '0x12a3a66720dD925fa93f7C895bC20Ca9560AdFe7',
+    ETH: '0xC61bA16e864eFbd06a9fe30Aab39D18B8F63710a',
+    USDC: '0x5D13179c5fa40b87D53Ff67ca26245D3D5B2F872',
   },
-}))
+}
 
-const Stake: FC = () => {
-  const styles = useStyles()
+const rewardTokenAddresses = {
+  WMATIC: '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270',
+  GNO: '0x9C58BAcC331c9aa871AFD802DB6379a98e80CEdb',
+}
 
+const Stake: FC = () => {
   const { bridges, sdk, networks, user, tokens } = useApp()
   const { provider } = useWeb3Context()
 
-  const stakingContracts = {
-    USDC: '0x2C2Ab81Cf235e86374468b387e241DF22459A265',
-    USDT: '0x07932e9A5AB8800922B2688FB1FA0DAAd8341772',
-    MATIC: '0x7dEEbCaD1416110022F444B03aEb1D20eB4Ea53f',
-    ETH: '0x7bceda1db99d64f25efa279bb11ce48e15fda427',
-  }
+  const availableNetworks = networks.filter(n =>
+    [ChainSlug.Gnosis, ChainSlug.Polygon].includes(n.slug as ChainSlug)
```

Comment:
maybe these chain slugs can be dynamic based on the staking addresses config keys?

Reply:
👍

```js
const availableNetworks = networks.filter(n =>
  Object.keys(stakingRewardsContracts).includes(n.slug as ChainSlug)
)
```